### ISSUE TITLE
[ads] Confirmation `conversion.action` is lost for non-rewards users

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#36614]
## Steps to Reproduce

1. Clean profile
2. View a search ad
3. Convert a search ad
4. Close the browser just after `Process conversion confirmation for search_result_ad with transaction id 8812f0e3-a5d0-4aeb-a366-5cecd823fe1d and creative instance id 788a9319-9ff9-4cf8-944b-98dcdc3dbfa8 in`
5. Run Browser
6. Check the conversion confirmation payload

## Actual result:

`conversion.action` is missing from  the payload
```
Content: {"creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","transactionId":"29db6808-61b0-4f1d-a247-48d958e9a3f1","type":"conversion"}
```

## Expected result:
`conversion.action` is included in the payload
```
Content: {"conversion":[{"action":"view"}],"creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","transactionId":"49fd2d95-68e8-47d7-9d23-1e21e532ba2f","type":"conversion"}
```

## Reproduces how often: 

Easily reproduced

## Brave version (brave://version info)

Brave | 1.63.169 Chromium: 122.0.6261.111 (Official Build) (64-bit)
-- | --
Revision | 0e0a975cf46d4d026e184ca61788e1da6f0d76bc
OS | Linux

cc @tmancey @iambrianfung @aseren

**Comment by kjozwiak** [comment]
Closing as the above has been fixed by https://github.com/brave/brave-core/pull/22529 and merged directly into `1.64.x`.

**Comment by btlechowski** [comment]
Verified with

Brave | 1.65.75 Chromium: 123.0.6312.29 (Official Build) nightly (64-bit)
-- | --
Revision | b36a621606293cdcba78f0d347e395cbe1d6986b
OS | Linux

Verified test plan from https://github.com/brave/brave-browser/issues/36614

Conversion confirmation was scheduled 
```
[7550:7550:0315/124526.317061:VERBOSE1:confirmations.cc(90)] Successfully added conversion confirmation for search_result_ad with transaction id 71dc68c9-6255-4f3d-978c-29781b349706 and creative instance id 788a9319-9ff9-4cf8-944b-98dcdc3dbfa8 to the queue
[7550:7550:0315/124526.322830:VERBOSE1:confirmations.cc(101)] Process conversion confirmation for search_result_ad with transaction id 71dc68c9-6255-4f3d-978c-29781b349706 and creative instance id 788a9319-9ff9-4cf8-944b-98dcdc3dbfa8 in 0 hours, 0 minutes, 18 seconds at 12:45:44.317

```

Verified confirmation queue holds conversion.action
```
{"queue":[{"ad_type":"search_result_ad","created_at":"13354976726312994","creative_instance_id":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","transaction_id":"71dc68c9-6255-4f3d-978c-29781b349706","type":"conversion","user_data":{"conversion":[{"action":"view"}]}}]}
```

Verified after restart conversion.action was included
```
[7957:7957:0315/124803.582396:VERBOSE1:redeem_non_reward_confirmation.cc(57)] Redeem non-reward confirmation
[7957:7957:0315/124803.582781:VERBOSE1:redeem_non_reward_confirmation.cc(66)] Create non-reward confirmation
[7957:7957:0315/124803.582964:VERBOSE6:redeem_non_reward_confirmation.cc(71)] URL Request:
  URL: https://search.anonymous.ads.bravesoftware.com/v3/confirmation/71dc68c9-6255-4f3d-978c-29781b349706
  Content: {"conversion":[{"action":"view"}],"creativeInstanceId":"788a9319-9ff9-4cf8-944b-98dcdc3dbfa8","transactionId":"71dc68c9-6255-4f3d-978c-29781b349706","type":"conversion"}

```
